test/plugin/test_out_elasticsearch.rb

```diff
@@ -954,10 +1001,10 @@ def test_connection_failed_retry
       raise Faraday::ConnectionFailed, "Test message"
     end
 
-    driver.emit(sample_record)
-
-    assert_raise(Fluent::ElasticsearchOutput::ConnectionFailure) {
-      driver.run
+    assert_raise(Fluent::Plugin::ElasticsearchOutput::ConnectionFailure) {
```

Comment:
I was able to get the test to pass by flushing immediately, not shutting down, and sleeping for as long as it takes all the retries to go throw:

```
assert_raise(Fluent::Plugin::ElasticsearchOutput::ConnectionFailure) {
  driver.run(default_tag: 'test', flush: true, shutdown: false) do
    driver.feed(sample_record)
  end
  sleep 6
}
```

Does this seem right to you @cosmo0920 ?